File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        output = [0]
        for i in range(1,n+1):
            val = i
            count = 0
            while val:
                val &= val - 1
                count += 1
            output.append(count)
        return output

        

File: Data Structures & Algorithms/non-cyclical-number/submission-0.js
class Solution {
    /**
     * @param {number} n
     * @return {boolean}
     */
    isHappy(n) {
        let slow = n;
        let fast = this.sumOfSquares(n);

        while (slow !== fast) {
            fast = this.sumOfSquares(fast);
            fast = this.sumOfSquares(fast);
            slow = this.sumOfSquares(slow);
        }

        return fast === 1;
    }

    /**
     * @param {number} n
     * @return {number}
     */
    sumOfSquares(n) {
        let output = 0;
        while (n !== 0) {
            output += (n % 10) ** 2;
            n = Math.floor(n / 10);
        }
        return output;
    }
}